test_queue_full fails intermittently

See [this run where test_queue_full fails](https://travis-ci.org/cherrypy/cherrypy/jobs/186442965) with a socket.timeout.

```
___________________ LimitedRequestQueueTests.test_queue_full ___________________

self = <cherrypy.test.test_conn.LimitedRequestQueueTests testMethod=test_queue_full>

    def test_queue_full(self):

        conns = []

        overflow_conn = None

    

        try:

            # Make 15 initial requests and leave them open, which should use

            # all of wsgiserver's WorkerThreads and fill its Queue.

            for i in range(15):

                conn = self.HTTP_CONN(self.HOST, self.PORT)

                conn.putrequest('POST', '/upload', skip_host=True)

                conn.putheader('Host', self.HOST)

                conn.putheader('Content-Type', 'text/plain')

                conn.putheader('Content-Length', '4')

                conn.endheaders()

                conns.append(conn)

    

            # Now try a 16th conn, which should be closed by the server immediately.

            overflow_conn = self.HTTP_CONN(self.HOST, self.PORT)

            # Manually connect since httplib won't let us set a timeout

            for res in socket.getaddrinfo(self.HOST, self.PORT, 0,

                                          socket.SOCK_STREAM):

                af, socktype, proto, canonname, sa = res

                overflow_conn.sock = socket.socket(af, socktype, proto)

                overflow_conn.sock.settimeout(5)

                overflow_conn.sock.connect(sa)

                break

    

            overflow_conn.putrequest('GET', '/', skip_host=True)

            overflow_conn.putheader('Host', self.HOST)

            overflow_conn.endheaders()

            response = overflow_conn.response_class(overflow_conn.sock, method='GET')

            try:

>               response.begin()

cherrypy/test/test_conn.py:812: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <http.client.HTTPResponse object at 0x7f4a3063a290>

    def begin(self):

        if self.headers is not None:

            # we've already started reading the response

            return

    

        # read until we get a non-100 response

        while True:

>           version, status, reason = self._read_status()

/opt/python/3.3.5/lib/python3.3/http/client.py:358: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <http.client.HTTPResponse object at 0x7f4a3063a290>

    def _read_status(self):

>       line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")

/opt/python/3.3.5/lib/python3.3/http/client.py:320: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <socket.SocketIO object at 0x7f4a3063a250>

b = <memory at 0x7f4a3469ce20>

    def readinto(self, b):

        """Read up to len(b) bytes into the writable buffer *b* and return

            the number of bytes read.  If the socket is non-blocking and no bytes

            are available, None is returned.

    

            If *b* is non-empty, a 0 return value indicates that the connection

            was shutdown at the other end.

            """

        self._checkClosed()

        self._checkReadable()

        if self._timeout_occurred:

            raise IOError("cannot read from timed out object")

        while True:

            try:

>               return self._sock.recv_into(b)

E               socket.timeout: timed out

/opt/python/3.3.5/lib/python3.3/socket.py:297: timeout

During handling of the above exception, another exception occurred:

self = <cherrypy.test.test_conn.LimitedRequestQueueTests testMethod=test_queue_full>

    def test_queue_full(self):

        conns = []

        overflow_conn = None

    

        try:

            # Make 15 initial requests and leave them open, which should use

            # all of wsgiserver's WorkerThreads and fill its Queue.

            for i in range(15):

                conn = self.HTTP_CONN(self.HOST, self.PORT)

                conn.putrequest('POST', '/upload', skip_host=True)

                conn.putheader('Host', self.HOST)

                conn.putheader('Content-Type', 'text/plain')

                conn.putheader('Content-Length', '4')

                conn.endheaders()

                conns.append(conn)

    

            # Now try a 16th conn, which should be closed by the server immediately.

            overflow_conn = self.HTTP_CONN(self.HOST, self.PORT)

            # Manually connect since httplib won't let us set a timeout

            for res in socket.getaddrinfo(self.HOST, self.PORT, 0,

                                          socket.SOCK_STREAM):

                af, socktype, proto, canonname, sa = res

                overflow_conn.sock = socket.socket(af, socktype, proto)

                overflow_conn.sock.settimeout(5)

                overflow_conn.sock.connect(sa)

                break

    

            overflow_conn.putrequest('GET', '/', skip_host=True)

            overflow_conn.putheader('Host', self.HOST)

            overflow_conn.endheaders()

            response = overflow_conn.response_class(overflow_conn.sock, method='GET')

            try:

                response.begin()

            except socket.error as exc:

                if exc.args[0] in socket_reset_errors:

                    pass # Expected.

                else:

                    tmpl = (

                        'Overflow conn did not get RST. '

                        'Got {exc.args!r} instead'

                    )

>                   raise AssertionError(tmpl.format(**locals()))

E                   AssertionError: Overflow conn did not get RST. Got ('timed out',) instead

cherrypy/test/test_conn.py:821: AssertionError

During handling of the above exception, another exception occurred:

self = <cherrypy.test.test_conn.LimitedRequestQueueTests testMethod=test_queue_full>

    def test_queue_full(self):

        conns = []

        overflow_conn = None

    

        try:

            # Make 15 initial requests and leave them open, which should use

            # all of wsgiserver's WorkerThreads and fill its Queue.

            for i in range(15):

                conn = self.HTTP_CONN(self.HOST, self.PORT)

                conn.putrequest('POST', '/upload', skip_host=True)

                conn.putheader('Host', self.HOST)

                conn.putheader('Content-Type', 'text/plain')

                conn.putheader('Content-Length', '4')

                conn.endheaders()

                conns.append(conn)

    

            # Now try a 16th conn, which should be closed by the server immediately.

            overflow_conn = self.HTTP_CONN(self.HOST, self.PORT)

            # Manually connect since httplib won't let us set a timeout

            for res in socket.getaddrinfo(self.HOST, self.PORT, 0,

                                          socket.SOCK_STREAM):

                af, socktype, proto, canonname, sa = res

                overflow_conn.sock = socket.socket(af, socktype, proto)

                overflow_conn.sock.settimeout(5)

                overflow_conn.sock.connect(sa)

                break

    

            overflow_conn.putrequest('GET', '/', skip_host=True)

            overflow_conn.putheader('Host', self.HOST)

            overflow_conn.endheaders()

            response = overflow_conn.response_class(overflow_conn.sock, method='GET')

            try:

                response.begin()

            except socket.error as exc:

                if exc.args[0] in socket_reset_errors:

                    pass # Expected.

                else:

                    tmpl = (

                        'Overflow conn did not get RST. '

                        'Got {exc.args!r} instead'

                    )

                    raise AssertionError(tmpl.format(**locals()))

            except BadStatusLine:

                # This is a special case in OS X. Linux and Windows will

                # RST correctly.

                assert sys.platform == 'darwin'

            else:

                raise AssertionError('Overflow conn did not get RST ')

        finally:

            for conn in conns:

>               conn.send(ntob('done'))

cherrypy/test/test_conn.py:830: 

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <http.client.HTTPConnection object at 0x7f4a3063a190>, data = b'done'

    def send(self, data):

        """Send `data' to the server.

            ``data`` can be a string object, a bytes object, an array object, a

            file-like object that supports a .read() method, or an iterable object.

            """

    

        if self.sock is None:

            if self.auto_open:

                self.connect()

            else:

                raise NotConnected()

    

        if self.debuglevel > 0:

            print("send:", repr(data))

        blocksize = 8192

        if hasattr(data, "read") :

            if self.debuglevel > 0:

                print("sendIng a read()able")

            encode = False

            try:

                mode = data.mode

            except AttributeError:

                # io.BytesIO and other file-like objects don't have a `mode`

                # attribute.

                pass

            else:

                if "b" not in mode:

                    encode = True

                    if self.debuglevel > 0:

                        print("encoding file using iso-8859-1")

            while 1:

                datablock = data.read(blocksize)

                if not datablock:

                    break

                if encode:

                    datablock = datablock.encode("iso-8859-1")

                self.sock.sendall(datablock)

            return

        try:

>           self.sock.sendall(data)

E           ConnectionResetError: [Errno 104] Connection reset by peer

/opt/python/3.3.5/lib/python3.3/http/client.py:875: ConnectionResetError
```

--- a/cherrypy/test/test_conn.py
+++ b/cherrypy/test/test_conn.py
@@ -7,8 +7,6 @@ import time
 import urllib.parse
 from http.client import BadStatusLine, HTTPConnection, NotConnected
 
-import pytest
-
 from cheroot.test import webtest
 
 import cherrypy

--- a/cherrypy/test/test_conn.py
+++ b/cherrypy/test/test_conn.py
@@ -778,7 +778,6 @@ socket_reset_errors += [
 class LimitedRequestQueueTests(helper.CPWebCase):
     setup_server = staticmethod(setup_upload_server)
 
-    @pytest.mark.xfail(reason='#1535')
     def test_queue_full(self):
         conns = []
         overflow_conn = None

